tests/categorical_test.py

```diff
@@ -0,0 +1,476 @@
+"""Tests for `categorical.py`."""
+
+from unittest import TestCase
+
+import equinox as eqx
+import jax
+import jax.numpy as jnp
+import numpy as np
+from parameterized import parameterized  # type: ignore
+
+from distreqx.distributions import categorical
+from distreqx.utils import math
+
+
+class CategoricalTest(TestCase):
+    def setUp(self):
+        self.dist = categorical.Categorical
+        self.key = jax.random.PRNGKey(0)
+
+    @parameterized.expand(
+        [
+            ("1d probs", (4,), True),
+            ("1d logits", (4,), False),
+            ("2d probs", (3, 4), True),
```

Comment:
Related to an early comment, is the 3 here just an implicit batch? Like if we have event size == () and this array doesn't get flattened, then its just 3 separate () sized events?